hs/app/reach-cli/Main.hs

```diff
@@ -0,0 +1,352 @@
+cmdReact :: Mod CommandFields Cli
+cmdReact = command "react" $ info (pure React) desc where
+  desc = progDesc "Run a simple React app"
+
+
+react :: Script ()
+react = undefined
+
+
+--------------------------------------------------------------------------------
+cmdRpcServer :: Mod CommandFields Cli
+cmdRpcServer = command "rpc-server" $ info (pure RpcServer) desc where
+  desc = progDesc "Run a simple Reach RPC server"
+
+
+rpcServer :: Script ()
+rpcServer = undefined
+
+
+--------------------------------------------------------------------------------
+cmdRpcRun :: Mod CommandFields Cli
+cmdRpcRun = command "rpc-run" $ info (pure RpcRun) desc where
+  desc = progDesc "Run an RPC server + frontend with development configuration"
+
+
+rpcRun :: Script ()
+rpcRun = undefined
+
+
+--------------------------------------------------------------------------------
+cmdDevnet :: Mod CommandFields Cli
+cmdDevnet = command "devnet" $ info (pure Devnet) desc where
+  desc = progDesc "Run only the devnet"
+
+
+devnet :: Script ()
+devnet = undefined
+
+
+--------------------------------------------------------------------------------
+cmdUpgrade :: Mod CommandFields Cli
+cmdUpgrade = command "upgrade" $ info (pure Upgrade) desc where
+  desc = progDesc "Upgrade Reach"
+
+
+upgrade :: Script ()
+upgrade = undefined
+
+
+--------------------------------------------------------------------------------
+cmdUpdate :: Mod CommandFields Cli
+cmdUpdate = command "update" $ info (pure Update) desc where
+  desc = progDesc "Update Reach Docker images"
+
+
+update :: Script ()
+update = undefined
+
+
+--------------------------------------------------------------------------------
+cmdDockerReset :: Mod CommandFields Cli
+cmdDockerReset = command "docker-reset" $ info (pure DockerReset) desc where
+  desc = progDesc "Docker kill and rm all images"
+
+
+dockerReset :: Script ()
+dockerReset = undefined
+
+
+--------------------------------------------------------------------------------
+cmdVersion :: Mod CommandFields Cli
+cmdVersion = command "version" $ info (pure Version) desc where
+  desc = progDesc "Display version"
+
+
+version :: Script ()
+version = cmd "echo" (quote $ "reach " <> reachVersion)
+
+
+--------------------------------------------------------------------------------
+cmdHelp :: Mod CommandFields Cli
+cmdHelp = command "help" $ info (pure Help) desc where
+  desc = progDesc "Show usage"
+
+
+help' :: Script ()
+help' = undefined
+
+
+--------------------------------------------------------------------------------
+cmdHashes :: Mod CommandFields Cli
+cmdHashes = command "hashes" $ info (pure Hashes) desc where
+  desc = progDesc "Display git hashes used to build each Docker image"
+
+
+hashes :: Script ()
+hashes = flip mapM_ reachImages $ \i -> do
+  let t = "reachsh/" <> i <> ":" <> reachVersion
+      s = cmd "docker" "run" "--entrypoint" "/bin/sh" t "-c" (quote "echo $REACH_GIT_HASH")
```

Comment:
Please follow our style with one binding per `let` unless you want mutually recursive bindings